lib/parsers/moov_parser/decoder.rb

```diff
@@ -130,6 +130,23 @@ def parse_stts_atom(io, _)
     stts
   end
 
+  def parse_stsd_atom(io, _)
+    version = read_byte_value(io)
+    is_v1 = version == 1
+    stsd = {
+      version: version,
+      flags: read_bytes(io, 3),
+      number_of_entries: is_v1 ? read_64bit_uint(io) : read_32bit_uint(io),
+      codecs: []
+    }
+    stsd[:number_of_entries].times {
+      codec_length = read_32bit_uint(io)
+      stsd[:codecs] << read_bytes(io, 4)
+      io.seek(io.pos + codec_length - 8) # 8 bytes is the header length containing the codec length and the codec name that we just did read
```

Comment:
Why do we have to seek? 🤔

Reply:
Because the "codec" itself is an atom containing lots of information inside that we don't care about. For example:
![image](https://user-images.githubusercontent.com/9882443/162049787-f7f7686f-682a-47e4-bd7a-9f562ee7801e.png)
As you can see in this example **hvc1** is the codec used, and also the info that we need, but beneath it has quite some info we don't care about, so I am skipping all of those to jump to the next codec, if there

Reply:
Right, now it makes sense. Thanks for the explanation!